sashimi-webapp/src/database/storage.js

```diff
@@ -0,0 +1,290 @@
+  }
+}
+
+function creationOfTables() {
+  if (typeof Promise === 'function') {
+    return new Promise((resolve, reject) => {
+      initCreateTable()
+      .then(() =>
+        query.isTableExistsInDatabase(constants.ENTITIES_USER)
+        .then(isUserTableFirstInitialize =>
+          createUserTable(isUserTableFirstInitialize)
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+          )
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() =>
+        query.isTableExistsInDatabase(constants.ENTITIES_ORGANIZATION)
+        .then(isOrganizationTableFirstInitialize =>
+          createOrganizationTable(isOrganizationTableFirstInitialize))
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() => query.isTableExistsInDatabase(constants.ENTITIES_FOLDER)
+        .then(isFolderTableFirstInitialize =>
+          createFolderTable(isFolderTableFirstInitialize))
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError))
+        .catch(sqlError => reject(sqlError))
+      )
+      .then(() => query.isTableExistsInDatabase(constants.ENTITIES_FILE_MANAGER)
+        .then(isFileTableFirstInitialize =>
+          createFileManagerTable(isFileTableFirstInitialize)
+          .then(isSuccess => isSuccess)
+          .catch(sqlError => reject(sqlError)))
+        .catch(sqlError => reject(sqlError))
+      .then(isSuccess => resolve(isSuccess))
+      .catch(sqlError => reject(sqlError))
+      );
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+export default class storage {
+  static constructor() {}
+
+  static initializeDatabase() {
+    if (typeof Promise === 'function') {
+      return new Promise((resolve, reject) => {
+        entitiesCreator.initializeDatabase(databaseName)
+        .then(step1 => creationOfTables()
+          .then(isFirstInstance => isFirstInstance)
+          .catch(sqlErr => reject(sqlErr)))
+        .then((isFirstInstance) => {
+          if (typeof isFirstInstance === 'boolean') {
+            return entitiesCreator.fillUpDefaultData()
+            .then(isFilledUp => resolve(isFilledUp))
+            .catch(sqlErr => reject(sqlErr));
+          } else {
+            return resolve(true);
+          }
+        })
+        .catch(sqlError => reject(sqlError));
+      });
+    } else {
+      throw new exceptions.PromiseFunctionNotDefined();
+    }
+  }
+
+  static loadAllFilesAndFolders() {
+    if (typeof Promise === 'function') {
+      return new Promise((resolve, reject) =>
+        query.getAllFilesAndFolders()
+        .then(returnedArr => resolve(returnedArr))
+        .catch(sqlError => reject(sqlError))
+      );
+    } else {
+      throw new exceptions.PromiseFunctionNotDefined();
+    }
+  }
+
+  static changeDatabaseName(newDatabaseName) {
```

Comment:
`changeDatabaseName()` may not be necessary to export as `filemanager` component will not be changing database name.